tests/db.rs

```diff
@@ -0,0 +1,184 @@
+    let writer_handles = (0..NUM_WRITERS)
+        .map(|writer_id| {
+            let db = db.clone();
+            let stop_writers = stop.clone();
+            tokio::spawn(async move {
+                // Get a random key in the keyspace. Don't use nth. Pick one at random.
+                let key = zero_pad_key(writer_id.try_into().unwrap(), KEY_LENGTH);
+                for i in 1..=WRITES_PER_TASK {
+                    if stop_writers.load(Ordering::Relaxed) {
+                        break;
+                    }
+
+                    // Write the incremented value
+                    db.put_with_options(
+                        &key,
+                        i.to_be_bytes().as_ref(),
+                        &PutOptions::default(),
+                        &WriteOptions {
+                            await_durable: false,
+                        },
+                    )
+                    .await
+                    .expect("Failed to write value");
+
+                    if i % 1000 == 0 {
+                        println!("Writer {} wrote {} values", writer_id, i);
+                    }
+                }
+            })
+        })
+        .collect::<Vec<_>>();
+
+    let stop_flushers = stop.clone();
+    let flusher_db = db.clone();
+    let flusher = tokio::spawn(async move {
+        while !stop_flushers.load(Ordering::Relaxed) {
+            flusher_db.flush().await.expect("Failed to flush");
+            tokio::time::sleep(Duration::from_millis(10)).await;
+        }
+    });
+
+    // Reader tasks: each reader reads all keys and verifies values are increasing
+    let reader_handles = (0..NUM_READERS)
+        .map(|reader_id| {
+            let db = db.clone();
+            let stop_readers = stop.clone();
+
+            tokio::spawn(async move {
+                let mut latest_values = HashMap::<usize, AtomicU64>::new();
+                let mut iterations = 0;
+
+                while !stop_readers.load(Ordering::Relaxed) {
+                    let mut rng = StdRng::from_entropy();
+                    // Pick a random key and validate that it's higher than the last value for that key
+                    let key = rng.gen_range(0..NUM_WRITERS);
+                    if let Some(bytes) = db
+                        .get(zero_pad_key(key.try_into().unwrap(), KEY_LENGTH))
+                        .await
+                        .expect("Failed to read value")
+                    {
+                        // Convert bytes to u64 value
+                        let value_bytes: [u8; 8] =
+                            bytes.as_ref().try_into().expect("Invalid value size");
+                        let current_value = u64::from_be_bytes(value_bytes);
+
+                        // Check if this value is greater than the last seen value for this key
+                        let last_seen_atomic = latest_values
+                            .entry(key)
+                            .or_insert(AtomicU64::new(0));
```

Comment:
Why does it insert `0` instead of `current_value`? It should still do a similar check with `0`, but wouldn't `current_value` me more accurate?

Reply:
Yea, that was from some AI gen'd code. Works either way, but I updated it to use current_value here: 648636f